game-core/build.gradle

```diff
@@ -301,7 +295,7 @@ gradle.taskGraph.whenReady { graph ->
 }
 
 check {
-    dependsOn 'test', 'integTest'
+    dependsOn 'integTest'
```

Comment:
you've double checked that 'test' is dependent on 'integTest'? To some extent I like the explicit dependency. :shrug:

Reply:
check is always dependend on test
See https://docs.gradle.org/current/userguide/java_plugin.html#_lifecycle_tasks

Reply:
Is that the case even if we explicitly declare the 'dependsOn'? I think 'dependsOn' will overwrite instead of appending dependencies. I could easily be wrong there though.

Reply:
It is (you can check what tasks would be executed by running `./gradlew --dry-run check`)

Reply:
[`Task#dependsOn()`](https://docs.gradle.org/current/javadoc/org/gradle/api/Task.html#dependsOn-java.lang.Object...-) adds dependencies:

```groovy
task.dependsOn 'otherTask1', 'otherTask2'
```

[`Task#setDependsOn()`](https://docs.gradle.org/current/javadoc/org/gradle/api/Task.html#setDependsOn-java.lang.Iterable-) overwrites dependencies:

```groovy
task.dependsOn = ['otherTask1', 'otherTask2']
```